config/tconfig.hpp

```diff
@@ -29,8 +29,72 @@
 #define MFEM_ALWAYS_INLINE
 #endif
 
-#define MFEM_TEMPLATE_BLOCK_SIZE 4
+// --- MFEM_VECTORIZE_LOOP (disabled)
+#if (__cplusplus >= 201103L) && !defined(MFEM_DEBUG) && defined(__GNUC__)
+//#define MFEM_VECTORIZE_LOOP _Pragma("GCC ivdep")
+#define MFEM_VECTORIZE_LOOP
+#else
+#define MFEM_VECTORIZE_LOOP
+#endif
+
+// --- MFEM_ALIGN_AS
+#if (__cplusplus >= 201103L)
+#define MFEM_ALIGN_AS(bytes) alignas(bytes)
+#elif !defined(MFEM_DEBUG) && (defined(__GNUC__) || defined(__clang__))
+#define MFEM_ALIGN_AS(bytes) __attribute__ ((aligned (bytes)))
+#else
+#define MFEM_ALIGN_AS(bytes)
+#endif
+
+// --- AutoSIMD or intrinsics
+#ifndef MFEM_USE_SIMD
+#include "simd/auto.hpp"
+#else
+#if defined(__VSX__)
+#include "simd/vsx.hpp"
+#elif defined (__bgq__)
+#include "simd/qpx.hpp"
+#elif defined(__x86_64__)
+#include "simd/x86.hpp"
+#else
+#error Unknown SIMD architecture
+#endif
+#endif
+
+// --- SIMD and BLOCK sizes
+#if defined(_WIN32)
+#define MFEM_SIMD_SIZE 8
+#define MFEM_TEMPLATE_BLOCK_SIZE 1
+#elif defined(__VSX__)
+#define MFEM_SIMD_SIZE 16
+#define MFEM_TEMPLATE_BLOCK_SIZE 2
+#elif defined(__x86_64__)
 #define MFEM_SIMD_SIZE 32
+#define MFEM_TEMPLATE_BLOCK_SIZE 4
```

Comment:
When `MFEM_USE_SIMD` is defined, these values should be adjusted: e.g. for `avx512` we want `MFEM_SIMD_SIZE` to be `64` bytes, right?

Reply:
`done`